cmd/influxd/run.go

```diff
@@ -171,10 +206,26 @@ func initServer(s *influxdb.Server, b *messaging.Broker) {
 	}
 }
 
+// joins a server to an existing cluster.
+func joinServer(s *influxdb.Server, u *url.URL, joinURLs []*url.URL) {
+	// TODO: Use separate broker and data join urls.
+
+	// Create data node on an existing data node.
+	for _, joinURL := range joinURLs {
+		if err := s.Join(u, joinURL); err != nil {
+			log.Printf("join: failed to connect data node: %s: %s", u, err)
+		} else {
+			log.Printf("join: connected data node to %s", u)
+			return
+		}
+	}
+	log.Fatalf("join: failed to connect data node to any specified server")
+}
+
 // opens the messaging client and attaches it to the server.
-func openServerClient(s *influxdb.Server, brokerURLs []*url.URL) {
+func openServerClient(s *influxdb.Server, joinURLs []*url.URL) {
```

Comment:
I do like this rename of the variable to `joinURLs`. Makes the code a little easier to follow.